### ISSUE TITLE
mu find seems to index only the first to/cc recipient

### ISSUE THREAD

**Issue by anadrome** [djcb/mu#826]
## Expected or desired behavior

For emails with multiple To: or Cc: recipients, I'd expect searches not to be sensitive to the order of the recipients. For example, in a search such as:

`mu find maildir:/example AND to:colleague@example.com`

I'd expect an email to show up if colleague@example is the first recipient, and also if they're the second.

## Actual behavior

For to: and cc: queries, messages are returned for `mu find` queries only when the searched-for recipient is the first recipient in the relevant field (i.e. first To: recipient, or first Cc: recipient).

I've resorted to doing something like this as a workaround, but it's not ideal,

`mu find maildir:/example --fields="t" | grep colleague@example.com`

## Versions of mu, mu4e/emacs, operating system etc.

mu 0.9.16, from Debian package maildir-utils version 0.9.16-1

**Comment by djcb** [comment]
I tried to reproduce this (and added a test-case), but didn't see the problem.  Can you provide a raw message (of course, anonymize any private information) where this occurs? Thanks.

**Comment by mmathis** [comment]
I got the impression that not all matching messages are returned when searching for recipients (using contact: or to:). I had a closer look at the raw messages which were not part of the search result and found a plausible explanation:
Only email addresses on the first line of To-header are considered. An example of a relevant line in the raw message header:
`
To: aaaaa bbbbbbbbbbbb <bbbbbbbbbbbb.aaaaa@cccccccccc.ddd>, "eeeeee
 ffffffffffff" <ffffffffffff.eeeeee@cccccccccc.ddd>, ggggggg hhhhhhh
	<hhhhhhh.ggggggg@cccccccccc.ddd>, iiiiii jjjjj <jjjjj@kkkkkkkkkkkk.ll>,
	mmmmmmm <mmmmmmm@kkkkkkkkkkkk.ll>
`
The message is contained in the search results when searching for "aaaaa" or "bbbbbbbbbbbb" and it isn't when searching for any other name (even not when searching for "eeeeee").

The number of characters, spaces and newlines of this example line are like in the original, only the characters have been replaced by the regular patterns.

I'm not sure whether this To-line is according to the specification but it was created by a mail program which is often used:
X-Mailer: Apple Mail (2.3124)
